javav2/example_code/s3/src/main/java/com/example/s3/express/S3DirectoriesScenario.java

```diff
@@ -0,0 +1,578 @@
+            }
+        } catch (CompletionException e) {
+            logger.error("Async operation failed: {} ", e.getCause().getMessage());
+            return;
+        }
+
+        logger.info("""
+            Notice how the normal bucket lists objects in lexicographical order, while the directory bucket does not. This is 
+            because the normal bucket considers the whole "key" to be the object identifier, while the directory bucket actually 
+            creates directories and uses the object "key" as a path to the object.
+            """);
+        waitForInputToContinue(scanner);
+    }
+
+    /**
+     * Demonstrates the performance difference between downloading an object from a directory bucket and a regular bucket.
+     *
+     * <p>This method:
+     * <ul>
+     *     <li>Prompts the user to choose the number of downloads (default is 1,000).</li>
+     *     <li>Downloads the specified object from the directory bucket and measures the total time.</li>
+     *     <li>Downloads the same object from the regular bucket and measures the total time.</li>
+     *     <li>Compares the time differences and prints the results.</li>
+     * </ul>
+     *
+     * <p>Note: The performance difference will be more pronounced if this example is run on an EC2 instance
+     * in the same Availability Zone as the buckets.
+     *
+     * @param bucketObject the name of the object to download
+     */
+    private static void demonstratePerformance(String bucketObject) {
+        logger.info("6. Demonstrate performance difference.");
+        logger.info("""
+            Now, let's do a performance test. We'll download the same object from each 
+            bucket 'downloads' times and compare the total time needed. Note: 
+            the performance difference will be much more pronounced if this
+            example is run in an EC2 instance in the same Availability Zone as 
+            the bucket.
+            """);
+        waitForInputToContinue(scanner);
+
+        int downloads = 1000; // Default value
+        logger.info("The number of downloads of the same object for this example is set at " + downloads + ".");
+
+        // Ask if the user wants to download a different number.
+        logger.info("Would you like to download a different number? (y/n): ");
+        String response = scanner.next().trim().toLowerCase();
+        if (response.equals("y")) {
+            int maxDownloads = 1_000_000;
+
+            // Ask for a valid number of downloads.
+            while (true) {
+                logger.info("Enter a number between 1 and " + maxDownloads + " for the number of downloads: ");
+                if (scanner.hasNextInt()) {
+                    downloads = scanner.nextInt();
+                    if (downloads >= 1 && downloads <= maxDownloads) {
+                        break;
+                    } else {
+                        logger.info("Please enter a number between 1 and " + maxDownloads + ".");
```

Comment:
This is a lot of logging. Perhaps emit a log message after each 50 (or 100) downloads. This may also speed things up a bit.